persistence/sql/persister_courier.go

```diff
@@ -170,3 +179,75 @@ func (p *Persister) IncrementMessageSendCount(ctx context.Context, id uuid.UUID)
 
 	return nil
 }
+
+func (p *Persister) FetchMessage(ctx context.Context, msgID uuid.UUID) (*courier.Message, error) {
+	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.FetchMessage")
+	defer span.End()
+
+	var message courier.Message
+	if err := p.GetConnection(ctx).
+		Where("id = ? AND nid = ?", msgID, p.NetworkID(ctx)).
+		Eager().
+		First(&message); err != nil {
+		return nil, sqlcon.HandleError(err)
+	}
+
+	return &message, nil
+}
+
+func (p *Persister) RecordDispatch(ctx context.Context, msgID uuid.UUID, status courier.CourierMessageDispatchStatus, err string) error {
+	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.RecordDispatch")
+	defer span.End()
+
+	dispatch := courier.MessageDispatch{
+		ID:        uuidx.NewV4(),
+		MessageID: msgID,
+		Error:     err,
+		Status:    status,
+		NID:       p.NetworkID(ctx),
+	}
+
+	if err := p.GetConnection(ctx).Create(&dispatch); err != nil {
+		return sqlcon.HandleError(err)
+	}
+
+	return nil
+}
+
+func (p *Persister) CreateTestData(ctx context.Context) error {
+
```

Comment:
I don't think this method belongs here. Why not put it into internal/testhelpers?

Reply:
I needed a way in cloud to generate a bunch of messages to test the pagination. So I added this + an endpoint to do that... Obviously wasn't going to propose it :D

This PR wasn't actually ready for PR yet, but I realized too late that it's not in Draft anymore. Sorry.